### PR TITLE
Completed initial implementation of add weight with imperial units

### PR DESCRIPTION
**Pull request by daljitcheema**

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class ProfilesController < ApplicationController
   before_action :set_profile
 
   def update
+    weight = Profile.convert_imperial(params[:stones], params[:pounds])
+    # @profile.update(profile_params)
+    @profile.update(weight: weight)
```
Comment (dmgarland) [inline app/controllers/profiles_controller.rb]: So note here that you're doing two saves to the database here - an `update` and a `save`, which would both generate update SQL statements. This also has the disadvantage that only weight can be updated.

File: app/views/profiles/edit.html.erb [hunk 2]
```diff
@@ -5,6 +5,15 @@
   <%= f.label :biography %>
   <%= f.text_field :biography %>
 
+  <hr>
+  <p>Unit: <%= select_tag("Weight-Unit"	, options_for_select(['Metric', "Imperial"], "Metric") ) %> </p>
+  Stones: <br>
+  <input type= "number" name="stones"><br>
+  Pounds: <br>
+  <input type= "number" name="pounds"><br>
```
Comment (dmgarland) [inline app/views/profiles/edit.html.erb]: How can I input metric then?

File: app/models/profile.rb [hunk 3]
```diff
@@ -2,4 +2,22 @@ class Profile < ActiveRecord::Base
...
+  def self.calculate_weight(params)
+    if params[:weight_unit] == 'Imperial'
+      return self.convert_imperial(params[:stones], params[:pounds])
+    else
+      return params[:weight]
+    end
+  end
+
+  def self.convert_imperial(stones, pounds)
+    weight = ((stones.to_f*14 + pounds.to_f)/2.2).round
```
Comment (dmgarland) [inline app/models/profile.rb]: put some whitespace around operators for readability please

File: app/views/profiles/edit.html.erb [hunk 4]
```diff
@@ -4,9 +4,24 @@
 
   <%= f.label :biography %>
   <%= f.text_field :biography %>
-
+<br>
```
Comment (dmgarland) [inline app/views/profiles/edit.html.erb]: BRs are banned!!!

File: app/views/profiles/edit.html.erb [hunk 5]
```diff
@@ -4,9 +4,24 @@
...
   <%= f.label :drinks, "Do you drink?" %>
   <%= f.check_box :drinks %>
+<hr>
+  <p>Unit: <%= select_tag("weight_unit" , options_for_select(['Metric', "Imperial"], "Metric") ) %> </p>
+
+  <div id="imperial">
+	 	Stones: <br>
```
Comment (dmgarland) [inline app/views/profiles/edit.html.erb]: Definition lists are good for things like this: dl, dt, dd

File: app/views/profiles/edit.html.erb [hunk 6]
```diff
@@ -4,9 +4,24 @@
...
+	  <input type= "number" name="stones"><br>
+	  Pounds: <br>
+	  <input type= "number" name="pounds"><br>
+	</div>
+
+	<div id="metric">
+		Kilograms:<br>
```
Comment (dmgarland) [inline app/views/profiles/edit.html.erb]: Indentation!

File: app/controllers/profiles_controller.rb (hunk 1 above)
Comment (daljitcheema) [inline app/controllers/profiles_controller.rb]: this has bee sorted by subsequent updates

File: app/views/profiles/edit.html.erb (hunk 2 above)
Comment (daljitcheema) [inline app/views/profiles/edit.html.erb]: resolved by having selective handling of both imperial and metric weight input

File: app/models/profile.rb (hunk 3 above)
Comment (daljitcheema) [inline app/models/profile.rb]: done

File: app/views/profiles/edit.html.erb (hunk 4 above)
Comment (daljitcheema) [inline app/views/profiles/edit.html.erb]: removed all BR's

File: app/views/profiles/edit.html.erb (hunk 5 above)
Comment (daljitcheema) [inline app/views/profiles/edit.html.erb]: done and used some of them

**Comment by daljitcheema** [comment]
resolved conflicts

**Comment by dmgarland** [comment]
This branch currently has failing tests. Please fix and update PR

```
undefined method `*' for nil:NilClass (ActionView::Template::Error)
      ./app/models/profile.rb:23:in `convert_metric'
      ./app/views/profiles/show.html.erb:8:in `_app_views_profiles_show_html_erb___3710258113108674200_70256931111180'
      ./features/step_definitions/search_gender_steps.rb:2:in `/^Bob is looking at his profile page$/'
      features/search_gender.feature:6:in `And Bob is looking at his profile page'
```

**Comment by dmgarland** [comment]
Are you running `rake` here? Still throws up errors for me...

**Comment by daljitcheema** [comment]
Resolved all outstanding issues, awaiting pull request to merge with Dev

**Pull request merged**